src/main/java/spoon/support/compiler/jdt/PositionBuilder.java

```diff
@@ -234,13 +235,51 @@ SourcePosition buildPositionCtElement(CtElement e, ASTNode node) {
 						bodyStart, bodyEnd,
 						lineSeparatorPositions);
 			}
+		} else if (e instanceof CtCatchVariable) {
+			Iterator<ASTPair> iterator = this.jdtTreeBuilder.getContextBuilder().stack.iterator();
+			iterator.next();
+			ASTPair next = iterator.next();
+			buildPositionCtElement(e, next.node);
+			sourceEnd = getSourceEndOfTypeReference(contents, (TypeReference) node, sourceEnd);
+			return cf.createSourcePosition(cu, sourceStart, sourceEnd, lineSeparatorPositions);
 		} else if (node instanceof TypeReference) {
 			sourceEnd = getSourceEndOfTypeReference(contents, (TypeReference) node, sourceEnd);
 		}
 
+		if (e instanceof CtModifiable) {
+			setModifiersPosition((CtModifiable) e, sourceStart, sourceEnd);
+		}
 		return cf.createSourcePosition(cu, sourceStart, sourceEnd, lineSeparatorPositions);
 	}
 
+
+	private void setModifiersPosition(CtModifiable e, int start, int end) {
+		CoreFactory cf = this.jdtTreeBuilder.getFactory().Core();
+		CompilationUnit cu = this.jdtTreeBuilder.getFactory().CompilationUnit().getOrCreate(new String(this.jdtTreeBuilder.getContextBuilder().compilationunitdeclaration.getFileName()));
+		CompilationResult cr = this.jdtTreeBuilder.getContextBuilder().compilationunitdeclaration.compilationResult;
+		char[] contents = cr.compilationUnit.getContents();
+
+		Set<CtExtendedModifier> modifiers = e.getExtendedModifiers();
+		if (start < 0 || end + 1 > contents.length) {
+			System.out.println(e);
+		}
+		String modifierContent = String.valueOf(Arrays.copyOfRange(contents, start, end + 1));
```

Comment:
I guess there is faster `new String(contents, start, end+1)` or something like that. Or why do we copy that array?